### PR TITLE
Porting cmd tests famework

### PR DESCRIPTION
**Pull request by sub-mod**
make test-cmd
to run tests under test folder

### CODE REVIEW

**Comment by elmiko** [comment]
i am having trouble getting this to run on the latest release of the 1.3 branch of openshift origin, logs: https://paste.fedoraproject.org/549917/14863946/

it also fails on 1.4, i'm not sure if the `openshift start` needs to be converted to `openshift start-build` though.

File: test/cmd/README.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+# OpenShift Command-Line Integration Test Suite
+
+This document describes how OpenShift developers should interact with the OpenShift command-line integration test suite.
+
+## Test Structure
+
+The script to run the entire suite lives in [`hack/test-cmd.sh`](./../../hack/test-cmd.sh). All of the test suites that make up
+the parent suite live in `test/cmd`, and are divided by functional area. 
+
+## Running Tests
```
Comment (elmiko) [inline test/cmd/README.md]: i think we should add a little section here on how to setup a test environment; must have openshift installed, should not have a cluster running, etc.

Comment (sub-mod) [inline test/cmd/README.md]: I will dockerize the test framework and re submit the PR.

**Comment by sub-mod** [comment]
strangely works for me on 1.3 release
https://paste.fedoraproject.org/549924/95038148/
Will Dockerize the test framework to avoid setup issues.

**Comment by elmiko** [comment]
that is really weird, i don't understand why it would fail with the same version on my machine. i'll try a clean build and run again.

**Comment by sub-mod** [comment]
just checked again , it seems to work on 1.4 as well https://paste.fedoraproject.org/549932/96075148/
Anyway i will dockerize to avoid more setup issues

**Comment by elmiko** [comment]
thanks subin, i think the issues i am seeing are related to locally configuration of the binaries produced by origin. i am re-testing now.

**Comment by elmiko** [comment]
ok, confirmed, tests are working for me locally.

**Comment by tmckayus** [comment]
I just ran this as a patch against the cli with the 1.4.1 vendor dir, it worked for me.  Any reason not to merge? (I just had to install a package to get the atomic-enterprise command)

**Comment by sub-mod** [comment]
it can't be integrated with travis.Since it needs openshift binary and also it creates openshift cluster and creates pods and then deletes them.Which again is not possible right now with travis.
This only good for local testing.

**Pull request merged**